src/bp/ui-admin/src/telemetry/client.ts

```diff
@@ -0,0 +1,162 @@
+import axios from 'axios'
+import { lang } from 'botpress/shared'
+import _ from 'lodash'
+import moment from 'moment'
+import ms from 'ms'
+import uuid from 'uuid'
+
+import store from '../store'
+
+import { EventData, EventPackageInfoType, StoreInfoType, TelemetryPackage } from './telemetry_type'
+
+const telemetryPackageVersion = '1.0.0'
+const dataClusterVersion = '1.0.0'
+
+export const endpoint = 'https://telemetry.botpress.dev'
+
+export const corsConfig = {
+  baseURL: endpoint,
+  headers: {
+    withCredentials: false
+  }
+}
+
+const storeInfos: StoreInfoType = {}
+
+const eventPackageInfo: EventPackageInfoType = {}
+
+const switchLock = (lockKey: string) => {
+  if (_.has(eventPackageInfo, lockKey)) {
+    eventPackageInfo[lockKey].locked = !eventPackageInfo[lockKey].locked
+  }
+}
```

Comment:
It is only used to switch the state of the lock of the package to send or sent. I should probably change the name of the parameter of this function because now that you mention it "lockKey" doesn't really mean anything

Reply:
Ok it seems that no one really understand why you do this locking logic, why is it necessary.

The only that's being done here is simply send data to an external endpoint why would this need some locking ?

Reply:
The lock is here to make sure that the package is not sent multiple time or not at the right time. In my code, I made sure that the ui_language package can only be sent every 8 hours. The lock is used to control the amount of info sent by events. I made this function to emulate a simple locking/unlocking mechanism

Reply:
Is it really necessary ? I find this over-complicated for nothing. The backend should avoid sending it multiple time; the client will only process this once.